Model/GuestPaymentInformationManagement.php

```diff
@@ -123,7 +123,7 @@ public function checkSpecificCountry($paymentMethod, $billingAddress)
         $configAllowSpecific = $this->configProviderMethodFactory->get($paymentMethodCode)->getAllowSpecific();
 
         if ($configAllowSpecific == 1) {
-            $countryId = $billingAddress->getCountryId();
+            $countryId = ($billingAddress != null ) ? $billingAddress->getCountryId() : null;
```

Comment:
it's not principal , but maybe it's better to place strict comparison with null in case if it's expected or (as for me) it's better to catch more cases by simple emptiness check